### PR TITLE
Cloudflare Worker Adapter

### PR DESCRIPTION
**Pull request by halfnelson**
An implementation of a Cloudflare Worker Site adapter

demo site running here: https://svelte-kit-demo.halfnelson.workers.dev

Not sure if this is wanted in the main repo, or if it can live in its own. Workers sites will probably be replaced by [Cloudflare Pages](https://blog.cloudflare.com/cloudflare-pages/) but those are static only for now.

If this ends up living in the main `kit` repo. We can add a github action to deploy the site to cloudflare when master is updated.

### CODE REVIEW

**Comment by antony** [comment]
This 100% belongs in the main repo. It's a prime example of how adaptable Svelte Kit is to different platforms.

File: packages/adapter-cloudflare-workers/README.md [hunk 1]
```diff
@@ -0,0 +1,27 @@
+# adapter-cloundflare-workers
+
+Adapter for Svelte apps that creates a cloudflare workers site, using a function for dynamic server rendering.
```
Comment (benmccann) [inline packages/adapter-cloudflare-workers/README.md:3]:
```suggestion
SvelteKit adapter that creates a Cloudflare Workers site using a function for dynamic server rendering.
```

File: packages/adapter-cloudflare-workers/files/render.js [hunk 2]
```diff
@@ -0,0 +1,70 @@
+import { render } from './app.js'; // eslint-disable-line import/no-unresolved
+import { getAssetFromKV, NotFoundError } from '@cloudflare/kv-asset-handler'; // eslint-disable-line import/no-unresolved
```
Comment (benmccann) [inline packages/adapter-cloudflare-workers/files/render.js:2]: any idea why the `eslint-disable-line` might be needed?

File: packages/adapter-cloudflare-workers/index.js [hunk 3]
```diff
@@ -0,0 +1,69 @@
...
+
+			builder.copy_static_files(bucket);
+			builder.copy_client_files(bucket);
+			builder.copy_server_files(entrypoint);
+
+			// copy the renderer
+			builder.copy(path.resolve(__dirname, 'files/render.js'), `${entrypoint}/index.js`);
+			builder.copy(path.resolve(__dirname, 'files/_package.json'), `${entrypoint}/package.json`);
+
+			// prerender
```
Comment (antony) [inline packages/adapter-cloudflare-workers/index.js:46]: Bit of a nitpick I know, but I think this comment is just noise, the function is called 'prerender' and the log message describes what it is doing :)

File: packages/adapter-cloudflare-workers/files/render.js (hunk 2 above)
Comment (halfnelson) [reply packages/adapter-cloudflare-workers/files/render.js:2]: yep. I don't particularly want to bundle up kv-asset-handler as part of the adapter, and the adapt function itself doesn't have a dependency on it, so lint complains.

File: packages/adapter-cloudflare-workers/index.js (hunk 3 above)
Comment (halfnelson) [reply packages/adapter-cloudflare-workers/index.js:46]: sure. will remove

**Comment by Rich-Harris** [comment]
this is awesome

**Comment by acoyfellow** [comment]
am i doing something wrong?
```
npm i @sveltejs/adapter-cloudflare-workers -D
npm ERR! code EUNSUPPORTEDPROTOCOL
npm ERR! Unsupported URL Type "workspace:": workspace:*
```

**Comment by halfnelson** [comment]
You need pnpm

**Comment by dummdidumm** [comment]
It's the same in all the other adapters, I wonder why it works there, then

**Comment by benmccann** [comment]
Hopefully fixed in https://github.com/sveltejs/kit/pull/749

**Comment by danawoodman** [comment]
Note to people who got here via Google like me, make sure to install with`@next` flag:

```shell
npm i -D @sveltejs/adapter-cloudflare-workers@next
```

**Pull request merged**